floats.go

```diff
@@ -69,9 +69,8 @@ func AddScaledTo(dst, y []float64, alpha float64, s []float64) []float64 {
 	if len(dst) != len(s) || len(dst) != len(y) {
 		panic("floats: lengths of slices do not match")
 	}
-	for i, val := range s {
-		dst[i] = y[i] + alpha*val
-	}
+	copy(dst, y)
```

Comment:
I assume there's no performance penalty?

Do you have a preference @kortschak?
On Jan 11, 2015 11:44 AM, "Fazlul Shahriar" <notifications@github.com>
wrote:

> In floats.go
> <https://github.com/gonum/floats/pull/30#discussion-diff-22769378>:
>
> > @@ -69,9 +69,8 @@ func AddScaledTo(dst, y []float64, alpha float64, s []float64) []float64 {
> >  	if len(dst) != len(s) || len(dst) != len(y) {
> >  		panic("floats: lengths of slices do not match")
> >  	}
> > -	for i, val := range s {
> > -		dst[i] = y[i] + alpha*val
> > -	}
> > +	copy(dst, y)
>
> If I'm understanding you correctly, there is no reason to have two
> versions of DaxpyUnitary. The blas/native code will call it as DaxpyUnitary(alpha,
> x, y, y) and the code here will call it as DaxpyUnitary(alpha, x, y, z).
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/gonum/floats/pull/30/files#r22769378>.
>